### ISSUE TITLE
segfault in cisd calculation

### ISSUE THREAD

**Issue by susilehtola** [psi4/psi4#606]
```
molecule {
0 1
symmetry c1
Mn -0.0128264866 -0.0002779912 -0.1091894544
O  -1.1037647823  1.0246639195 -1.2821459116
O  -0.4319309790 -1.5910632792 -1.0423314985
N   0.7243918073  1.6205280457  0.7709992473
N   1.3008767173 -0.9287768925  1.0635839133
C  -1.2886186814  2.2834208702 -1.2181338272
C  -0.7050865946  3.1821062059 -0.3604271375
C   0.3534920334  2.8195308082  0.5207059889
C   0.2177419816 -2.6945497951 -0.9662726987
C   1.2181378357 -3.0250490115 -0.0961995911
C   1.7180430400 -2.1229929449  0.8899899444
C   1.9393113446  1.3525171542  1.5367699106
C   1.8333722479 -0.0604962603  2.1055467704
H   0.8908035853  3.6241374839  1.0039698993
H   2.5024254412 -2.4854165868  1.5399007635
H   2.7858100041  1.4098930461  0.8661311665
H   1.1490336627 -0.0652673733  2.9502674019
H   2.0638511115  2.0769335299  2.3353274192
H   2.7993768711 -0.4085469833  2.4534782102
H  -1.9892347032  2.6686747336 -1.9454961675
H  -0.0986239324 -3.4441353556 -1.6764363233
Cl  1.8821942342  0.3098852687 -1.6599082362
O  -1.1395010178 -0.1703680057  0.9263551082
H  -0.9312248148  4.2251864392 -0.4696082573
H   1.6695378989 -3.9953914470 -0.1633815245
}

# Run SCF, initialize in smaller basis
set basis 6-31G*
set basis_guess 3-21G
set scf_type pk
set soscf true
scf_e, scf_wfn = energy('scf', return_wfn=True)

# Get CISD active space guess. 69 occupied orbitals, out of which 32
# are core electrons. Leaves No=37 orbitals, excitable into Nv=204
# virtuals. Let's take only the first 50 virtuals
set restricted_docc [32]
set restricted_uocc [154]
set qc_module detci
set nat_orbs true
ci_e, ci_wfn = energy('cisd', ref_wfn=scf_wfn, return_wfn=True)
```
segfaults after the 20 most important determinants have been written to output.

Backtrace gives
```
#0  __GI___libc_free (mem=mem@entry=0x550020205841) at malloc.c:2963
#1  0x00007f6080ce0a78 in psi::detci::CIWavefunction::print_vector (this=0x55948f151ec0, vec=std::shared_ptr (count 4, weak 0) 0x559528f71aa0, root=0)
    at /home/work/psi4/psi4.susi/psi4/src/psi4/detci/ciwave.cc:683
#2  0x00007f6080706283 in pybind11::cpp_function::cpp_function<void, psi::detci::CIWavefunction, std::shared_ptr<psi::detci::CIvect>, int, pybind11::name, pybind11::is_method, pybind11::sibling, char [10]>(void (psi::detci::CIWavefunction::*)(std::shared_ptr<psi::detci::CIvect>, int), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&, char const (&) [10])::{lambda(psi::detci::CIWavefunction*, std::shared_ptr<psi::detci::CIvect>, int)#1}::operator()(psi::detci::CIWavefunction*, std::shared_ptr<psi::detci::CIvect>, int) const (args#1=0,
    args#0=std::shared_ptr (count -555219596, weak 1224731640) 0x7ffef53db730, c=0x55948f151ec0, __closure=0x55948e9b84e8) at /home/work/psi4/install/include/pybind11/pybind11.h:62
#3  pybind11::detail::argument_loader<psi::detci::CIWavefunction*, std::shared_ptr<psi::detci::CIvect>, int>::call_impl<void, pybind11::cpp_function::cpp_function<void, psi::detci::CIWavefunction, std::shared_ptr<psi::detci::CIvect>, int, pybind11::name, pybind11::is_method, pybind11::sibling, char [10]>(void (psi::detci::CIWavefunction::*)(std::shared_ptr<psi::detci::CIvect>, int), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&, char const (&) [10])::{lambda(psi::detci::CIWavefunction*, std::shared_ptr<psi::detci::CIvect>, int)#1}&, 0ul, 1ul, 2ul>(pybind11::cpp_function::cpp_function<void, psi::detci::CIWavefunction, std::shared_ptr<psi::detci::CIvect>, int, pybind11::name, pybind11::is_method, pybind11::sibling, char [10]>(void (psi::detci::CIWavefunction::*)(std::shared_ptr<psi::detci::CIvect>, int), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&, char const (&) [10])::{lambda(psi::detci::CIWavefunction*, std::shared_ptr<psi::detci::CIvect>, int)#1}&, pybind11::detail::index_sequence<0ul, 1ul, 2ul>) (f=..., this=0x7ffef53db750)
    at /home/work/psi4/install/include/pybind11/cast.h:1277
```
so it seems to be crashing in the call
```
  free(mi_ibidx);
```

**Comment by susilehtola** [comment]
The crash is also reproducible with a slightly smaller input
```
molecule {
0 1
symmetry c1
Mn -0.0128264866 -0.0002779912 -0.1091894544
O  -1.1037647823  1.0246639195 -1.2821459116
O  -0.4319309790 -1.5910632792 -1.0423314985
N   0.7243918073  1.6205280457  0.7709992473
N   1.3008767173 -0.9287768925  1.0635839133
C  -1.2886186814  2.2834208702 -1.2181338272
C  -0.7050865946  3.1821062059 -0.3604271375
C   0.3534920334  2.8195308082  0.5207059889
C   0.2177419816 -2.6945497951 -0.9662726987
C   1.2181378357 -3.0250490115 -0.0961995911
C   1.7180430400 -2.1229929449  0.8899899444
C   1.9393113446  1.3525171542  1.5367699106
C   1.8333722479 -0.0604962603  2.1055467704
H   0.8908035853  3.6241374839  1.0039698993
H   2.5024254412 -2.4854165868  1.5399007635
H   2.7858100041  1.4098930461  0.8661311665
H   1.1490336627 -0.0652673733  2.9502674019
H   2.0638511115  2.0769335299  2.3353274192
H   2.7993768711 -0.4085469833  2.4534782102
H  -1.9892347032  2.6686747336 -1.9454961675
H  -0.0986239324 -3.4441353556 -1.6764363233
Cl  1.8821942342  0.3098852687 -1.6599082362
O  -1.1395010178 -0.1703680057  0.9263551082
H  -0.9312248148  4.2251864392 -0.4696082573
H   1.6695378989 -3.9953914470 -0.1633815245
}

# Run SCF, initialize in smaller basis
set basis 3-21G
set scf_type pk
set soscf true
scf_e, scf_wfn = energy('scf', return_wfn=True)

# Get CISD active space guess. 69 occupied orbitals, out of which 32
# are core electrons. Leaves No=37 orbitals, excitable into Nv=110
# virtuals. Let's take only the first 40 virtuals
set restricted_docc [32]
set restricted_uocc [70]
set qc_module detci
set nat_orbs true
ci_e, ci_wfn = energy('cisd', ref_wfn=scf_wfn, return_wfn=True)
```

**Comment by susilehtola** [comment]
Fixed by #607.
